### PR TITLE
Implemented a basic API for webapp developers.

### PR DESCRIPTION
**Pull request by sammyers**

### CODE REVIEW

File: library/Makefile [hunk 1]
```diff
@@ -1,23 +0,0 @@
...
-	gcc -c codes.c
-
-date.o: date.h date.c
-	gcc -c date.c
-
-server.o: common.h request.h response.h server.c
-	gcc -c server.c
-
-clean:
-	rm server server.o request.o response.o codes.o date.o
```
Comment (billmwong) [inline library/Makefile]: Do I have to make from the root now :sob:

File: library/request.c [hunk 2]
```diff
@@ -84,19 +86,19 @@ Request *build_request_from_socket(int socket)
...
-        char lineBuffer[LINE_BUFFER_SIZE];
-        read_socket_until_stopper(socket, '\r', lineBuffer);
+        char line_buffer[LINE_BUFFER_SIZE];
+        read_socket_until_stopper(socket, '\r', line_buffer);
 
         // If buffer is empty, we're done with headers
-        if (!lineBuffer[0]) break;
-        add_header_to_request(req, build_header_from_string(lineBuffer));
+        if (!line_buffer[0]) break;
+        add_header_to_request(req, build_header_from_string(line_buffer));
```
Comment (billmwong) [inline library/request.c]: you just like pooping on my code sam

File: library/Makefile (hunk 1 above)
Comment (sammyers) [inline library/Makefile]: Yes

File: library/request.c (hunk 2 above)
Comment (sammyers) [inline library/request.c]: rekt

**Pull request merged**